src/components/SelectionList/types.ts

```diff
@@ -73,6 +73,9 @@ type ListItem = {
     /** Whether this option is disabled for selection */
     isDisabled?: boolean | null;
 
+    /** To apply diabled style when item is not diabled to unselect */
+    applyDisabledStyle?: boolean | null;
```

Comment:
Why exactly we need separate `applyDisabledStyle` attribute. Can't we achieve the same result with `isDisabled`?

Reply:
After a lot of thinking and reviewing, I think this is not a good approach to this problem. However I didn't manage to come up with better solution yet. But I do think that we should really think about this solution before proceeding with it.

@dangrous What do you think about changes with `applyDisabledStyle`?

Reply:
Hi! I'm still partially on leave. Let me give this some thought, see if I can figure out something!

Reply:
Okay, well, I came up with nothing. Can you post about this in the Slack channel and see if anyone else has some clever ideas to cover all the situations? I also would love not to go this route since it seems messy - but if it's the only option, then we can do it.

Reply:
I think that sounds better to me, what do you think @alitoshmatov? Alternately, can we use an existing value of `isSelected` or similar? How do we know if it's currently selected right now?

Reply:
I agree, can we just check if currently selected item is disabled, if yes we would allow it to be deselected without introducing new attribute @Krishna2323

Reply:
Bump @Krishna2323 what do you think about this? Do you think you can work on the solution soon so we can get this updated? Thanks!